modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -784,11 +759,28 @@ object SeqexecEngine extends SeqexecConfiguration {
 
     val smartGcal = smartGcalEnable.fold(initSmartGCal(smartGCalHost, smartGCalDir), IO.unit)
 
-    smartGcal *>
-      epicsInit *>
-      (for {
-        now <- IO(LocalDate.now)
-      } yield Settings(site,
+    def gpiClient: cats.effect.Resource[IO, GpiClient[IO]] =
+      if (gpiControl === FullControl) {
+        GpiClient.gpiClient[IO](gpiUrl, ec)
+      } else {
+        GpiClient.simulatedGpiClient(ec)
+      }
+
+    def ghostClient: cats.effect.Resource[IO, GhostClient[IO]] =
+      if (gpiControl === FullControl) {
+        GhostClient.ghostClient[IO](ghostUrl, ec)
+      } else {
+        GhostClient.simulatedGhostClient(ec)
+      }
+
+    def settings: IO[Settings] = {
```

Comment:
This is a somewhat large change as instead of getting `IO[GpiClient]` we'll get `Resource[IO, GPIClient]` instead